cishouseholds/derive.py

```diff
@@ -10,6 +10,47 @@
 from pyspark.sql import Window
 
 
+def assign_multigen(
+    df: DataFrame,
+    column_name_to_assign: str,
+    participant_id_column,
+    hh_id_column: str,
+    visit_date_column: str,
+    dob_column: str,
+):
+    """ """
+    transformed_df = df.groupBy(hh_id_column, visit_date_column).count()
+    transformed_df = transformed_df.join(df.select(hh_id_column, participant_id_column, dob_column), on=hh_id_column)
+    transformed_df = assign_school_year_september_start(
+        df=transformed_df,
+        dob_column=dob_column,
+        visit_date_column=visit_date_column,
+        column_name_to_assign="school_year",
+    )
+    transformed_df = assign_age_at_date(
+        df=transformed_df,
+        column_name_to_assign="age_at_visit",
+        base_date=F.col(visit_date_column),
+        date_of_birth=F.col(dob_column),
+    )
+
+    gen1_flag = F.when((F.col("age_at_visit") > 49), 1).otherwise(0)
+    gen2_flag = F.when(
+        ((F.col("age_at_visit") <= 49) & (F.col("age_at_visit") >= 17)) | (F.col("school_year") >= 12), 1
+    ).otherwise(0)
+    gen3_flag = F.when((F.col("school_year") <= 11), 1).otherwise(0)
+
+    window = Window.partitionBy(hh_id_column, visit_date_column)
+    gen1_exists = F.when(F.sum(gen1_flag).over(window) >= 1, 1).otherwise(0)
```

Comment:
If these are changed to `bool` type (`True`), it will simplify the condition below to `F.when(gen1_exists & gen2_exists...)` to make it more readable